### PR TITLE
rain-bittorrent: init at 1.12.17

### PR DESCRIPTION
**Pull request by justinrubek**
## Description of changes

Added package https://github.com/cenkalti/rain

This package is a bittorrent client. It provides an RPC server that other applications can connect to in order to initiate a download as well as a command-line client in the same binary.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ra/rain-bittorrent/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "cenkalti";
+    repo = "rain";
+    rev = "v1.12.17";
+    sha256 = "sha256-ZO48hY1GnZiSN1LyQ821B1CDmRZJF2AFHBRrHbu07IQ=";
+  };
+
+  vendorHash = "sha256-NS/neEyFKYiK7BvYaDXqzNWkIDBw/p9FiAE3qDge6DU=";
+
+  meta = with lib; {
+    description = "🌧 BitTorrent client and library in Go";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ra/rain-bittorrent/package.nix:21]:
```suggestion
    description = "BitTorrent client and library in Go";
```

File: pkgs/by-name/ra/rain-bittorrent/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule {
+  pname = "rain";
+  version = "1.12.17";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ra/rain-bittorrent/package.nix:9]:
```suggestion
  version = "1.13.0";
```
New version has been released by now.

File: pkgs/by-name/ra/rain-bittorrent/package.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+}:
+
+buildGoModule {
+  pname = "rain";
+  version = "1.30.0";
```
Comment (FliegendeWurst) [inline pkgs/by-name/ra/rain-bittorrent/package.nix:9]: does not match the fetch below

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ra/rain-bittorrent/package.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  src = fetchFromGitHub {
+    owner = "cenkalti";
+    repo = "rain";
+    rev = "v1.13.0";
```
Comment (Aleksanaa) [inline pkgs/by-name/ra/rain-bittorrent/package.nix:14]:
```suggestion
    rev = "refs/tags/v${version}";
```

File: pkgs/by-name/ra/rain-bittorrent/package.nix [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "sha256-pz20vhr3idXja7wYIdVr1dosSpqYiQfeho66rqd718I=";
```
Comment (Aleksanaa) [inline pkgs/by-name/ra/rain-bittorrent/package.nix:15]:
```suggestion
    hash = "sha256-pz20vhr3idXja7wYIdVr1dosSpqYiQfeho66rqd718I=";
```

File: pkgs/by-name/ra/rain-bittorrent/package.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+  vendorHash = "sha256-40DK0D9TRJDfrMbBJNpcNzvjKb/uXN/Yz5Bb7oXBh+E=";
+
+  meta = with lib; {
+    description = "BitTorrent client and library in Go";
+    homepage = "https://github.com/cenkalti/rain";
+    license = licenses.mit;
+    maintainers = with maintainers; [ justinrubek ];
+    mainProgram = "rain";
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/ra/rain-bittorrent/package.nix:26]:
```suggestion
  meta = {
    description = "BitTorrent client and library in Go";
    homepage = "https://github.com/cenkalti/rain";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ justinrubek ];
    mainProgram = "rain";
  };
```

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**